### DISCUSSION TITLE
Export findings to CSV using the API

### DISCUSSION THREAD

**Discussion by SeaofThought** [DefectDojo/django-DefectDojo#9777] [Q&A]
Hi,

DefectDojo allows for the export of findings in CSV /  Excel format, as illustrated here:

https://defectdojo.github.io/django-DefectDojo/integrations/exporting/

However, I couldn't find a way to get the CSV files using the API.

Am I doing something wrong and, if not, is there a date for this feature to be implemented in the API?

Thanks.

**Comment by mtesauro** [comment]
@SeaofThought You're correct that the CSV / Excel download is a UI feature only. The reason for this is:

1. A JS library is used to produce those results which is why they are UI only
2. The API is a REST API which produces data representations in JSON. The assumption is that if you have the data in JSON (a machine readable format) then conversion to CSV or Excel is something that will happen at the client that requested the data. There's several libraries and other tools which can convert JSON to CSV or Excel if that's the final format you need the data in from the API.

I cannot see DefectDojo offering CSV for Excel files  from the API in the future for these reasons.
